portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,101 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const counterObj = {
+    picturesIndex: 0
+};
+const allPictures = ['20060919215839_DSC_6806.JPG', '20060919222117_DSC_6821.JPG', '20170204094939_IMG_1368.JPG'
+, '20170209104355_IMG_1531.JPG', '20170213143558_IMG_1735.JPG', '20170222180214_IMG_1834.JPG', '20170302135232_IMG_2101.JPG',
+'IMG_20170115_202802_153.jpg'];
+
+/** 
+* change the picture in the dom gallery according to pictureIndex variable
+ */
+function changePictureInDom()
+{
+    const pic = allPictures[counterObj.picturesIndex];
+    const imgContainer = document.getElementById('img-container');
+    imgContainer.src = '/images/' + pic;
+}
+
+/** 
+* moves the pictureIndex forward
+*/
+function nextPicture()
+{
+    counterObj.picturesIndex++;
+    counterObj.picturesIndex = counterObj.picturesIndex === allPictures.length? 0 : counterObj.picturesIndex;   
+    changePictureInDom();
+}
+
 /**
- * Adds a random greeting to the page.
+* moves the pictureIndex backwards
+*/
+function previousPicture()
+{
+    counterObj.picturesIndex--;
+    counterObj.picturesIndex = counterObj.picturesIndex < 0 ? allPictures.length - 1 : counterObj.picturesIndex;   
+    changePictureInDom();
+}
+
+/**
+* displays the haka videwin the iframe
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function displayHakaFrame()
+{
+    const hakaIframe = document.getElementById('hakaIframe');
+    hakaIframe.height = 345;
+    hakaIframe.width = 420;
+    hakaIframe.style.visibility = 'visible';
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    const hakaVideoButton = document.getElementById('hakaVideoButton')
+    hakaVideoButton.style.visibility = 'hidden';
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+/**
+* requests for the recommendations from db and display it on the dom
+ */
+function getComments(url)
+{
+    //default limit is 15
+    if (url === undefined) url = "/data?limit=15";
+    fetch(url).then(response => response.json()).then(comments => {
+        let commentsElement = document.getElementById('comments')
+
+        //clear previus comments
+        commentsElement.innerHTML = '';
+
+        comments.forEach(comment=> {
+            let com = document.createElement('li');
```

Comment:
I think it should work with const. 
In js the const is for the pointer on the object. There are other places in the file to apply this

Reply:
thanks, good to know- changed most the variable declarations in the file